eZ/Publish/Core/REST/Server/Service/RootResourceBuilder.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * File containing the resource service class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ *
+ * @version //autogentag//
+ */
+namespace eZ\Publish\Core\REST\Server\Service;
+
+use eZ\Publish\Core\REST\Common\Values;
+use eZ\Publish\Core\REST\Common\Values\Root;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\Routing\RouterInterface;
+
+class RootResourceBuilder
+{
+    /**
+     * @var \Symfony\Component\Routing\RouterInterface
+     */
+    protected $router;
+
+    /**
+     * @var \Symfony\Component\Routing\RouterInterface
+     */
+    protected $templateRouter;
+
+    /**
+     * @var array
+     */
+    protected $resourceConfig;
+
+    /**
+     * Creates a new resource builder.
+     *
+     * @param \Symfony\Component\Routing\RouterInterface $router
+     * @param \Symfony\Component\Routing\RouterInterface $templateRouter
+     * @param array $rest
+     */
+    public function __construct(RouterInterface $router, RouterInterface $templateRouter, array $rest)
```

Comment:
Why `$rest` as an array with a `resources` key ? It could be an array of options, including `'resources'`. But given that those resources are the base component of this builder, I'd rather call it `$resources`, and remove the level 1 array key